v2/internal/controllers/crd_redhatopenshift_openshiftcluster_20230401_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	aro "github.com/Azure/azure-service-operator/v2/api/redhatopenshift/v1api20231122"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+// TODO: To re-record this test, create a new service principal and save its details into the
+// TODO: ARO_CLIENT_SECRET env variable and replace clientID and principalId below
+// TODO: This is required because the ARO resource requires a service principal for input currently. Hopefully
+// TODO: we can revisit this in the future when they support other options.
+func Test_RedHatOpenShift_OpenShiftCluster_CRUD(t *testing.T) {
+	t.Parallel()
+
+	if *isLive {
+		t.Skip("ARO cluster test requires ServicePrincipal creation and referenced in the cluster object.")
+	}
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	secretName := "aro-secret"
+	secretKey := "client-secret"
+	// TODO: Replace the principalID, clientSecret and clientId vars below with principalId, clientSecret and clientId of your SP
+	principalId := "5c6be76c-5fc4-4817-992d-22027b44c402"
```

Comment:
I think these should be env variables too, or hardcoded to 00000's GUIDs?

Is the issue with changing these that the test recordings don't pass then?